qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -161,6 +464,206 @@ def abort(self):
+
+            abort_enabled: Boolean, enables or disables the automatic abort
+                function for the CV sweep measurement.
+
+            sweep_mode: Linear sweep, log sweep, linear 2 way sweep or
+                log 2 way sweep
+
+            volt_monitor: Boolean
+
+        """
+
+        self.root_instrument.enable_channels(self.channels)
+        self.adc_mode(adc_mode)
+        self.adc_coef(adc_coef)
+        self.frequency(freq)
+        self.voltage_ac(ac_rms)
+        self.cv_sweep.sweep_auto_abort(abort_enabled)
+        self.cv_sweep.post_sweep_voltage_cond(post_sweep_voltage_cond)
+        self.cv_sweep.hold(hold_delay)
+        self.cv_sweep.delay(delay)
+        self.cv_sweep.step_delay(step_delay)
+        self.cv_sweep.trigger_delay(trigger_delay)
+        self.cv_sweep.measure_delay(measure_delay)
+        self.sweep_mode(sweep_mode)
+        self.sweep_start(v_start)
+        self.sweep_end(v_end)
+        self._sweep_steps = n_steps
+        self.sweep_steps(self._sweep_steps)
+        self.measurement_mode(constants.MM.Mode.CV_DC_SWEEP)
+        self.impedance_model(imp_model)
+        self.ac_dc_volt_monitor(volt_monitor)
+        self.ranging_mode(ranging_mode)
+        self.measurement_range_for_non_auto(fixed_range_val)
+
+        err = self.root_instrument.error_message()
+        if err == '+0,"No Error."':
+            self.setup_fnc_already_run = True
+        return err
+
+    @staticmethod
+    def parse_sweep_data(raw_data: str) -> tuple:
+        no_commas = raw_data.split(',')
+        no_str = [float(val[3:]) for val in no_commas]
+        param1 = []
+        param2 = []
+        for i, val in enumerate(no_str):
+            if i % 2:
+                param2.append(val)
+            else:
+                param1.append(val)
+
+        param1 = list(np.array(param1))
+        param2 = list(np.array(param2))
+        return param1, param2
+
+
+class CVSweepMeasurement(MultiParameter):
+    """
+    CV sweep measurement outputs a list of primary (capacitance) and secondary
+    parameter (disipation).
+
+    Args:
+        name: Name of the Parameter.
+
+        instrument: Instrument to which this parameter communicates to.
+    """
+
+    def __init__(self, name, instrument, **kwargs):
+        super().__init__(
+            name,
+            names=tuple(['Capacitance', 'Dissipation']),
+            units=tuple(['F', 'unit']),
+            labels=tuple(['Parallel Capacitance', 'Dissipation factor']),
+            shapes=((1,),) * 2,
+            setpoint_names=(('Voltage',),) * 2,
+            setpoint_labels=(('Voltage',),) * 2,
+            setpoint_units=(('V',),) * 2,
+            **kwargs)
+        self._instrument = instrument
+
+    def get_raw(self):
+        if not self._instrument.setup_fnc_already_run:
+            raise Warning('Sweep setup has not yet been run successfully')
+
+        self.shapes = ((self._instrument.sweep_steps(),),) * 2
```

Comment:
Think how now to do this. How about if Multiparam took setpoint function as paramaterwithsetpoints?

Reply:
decided: to keep as is for now, but improve MultiParameter to use callables for shapes and setpoints, and then fix it here.